050-Manage/030-migrations.mdx

```diff
@@ -73,10 +73,131 @@ Migration files should not be modified. If externally modified, the record will
 
-In the editor you can provide content for the pgroll operations array and start a migration. Once submitted, the migration will enter the Active state and must be manually completed with the "Complete migration" action button before further schema edits can take place. While in Active state, the migration can also be rolled back.
+Multi-version schema migrations address the pain point of having to keep your application code in sync with your database schema by allowing you two active database schemas at once. With multi-version schema migrations, running a migration means being able to preview the new version of your schema alongside the old one. Reading and writing data in both schemas continues to work normally. As a result of having two active schema versions, rollbacks become seamless operations which is particularly useful in the case of an unwanted schema change or backfill.
+
+<Alert status="warning">
+  Currently multi-version schema migrations are only supported via the Migration Editor in the UI. Migrations made via
+  the Table View and Schema View will complete automatically.
+</Alert>
+
+![The Migration Editor](images/migrations-multiversion/migration-editor.png)
+
+Some common schema operations are:
+
+<TabbedCode tabs={['create_table', 'add_column', 'rename_table', 'alter_column', 'drop_table', 'drop_column']}>
+```jsonc
+[{
+  "create_table": {
+    "name": "customers",
+    "columns": [
+      {
+        "name": "id",
+        "type": "integer",
+        "pk": true
+      },
+    ]
+  }
+}]
+```
+```jsonc
+[{
+  "add_column": {
+    "table": "customers",
+    "up": "UPPER(name)",
+    "column": {
+      "name": "full_name",
+      "type": "varchar(255)",
+      "nullable": true
+    }
+  }
+}]
+```
+```jsonc
+[{
+  "rename_table": {
+    "from": "customers",
+    "to": "clients"
+  }
+}]
+```
+```jsonc
+[{
+  "alter_column": {
+    "table": "clients",
+    "column": "full_name",
+    "name": "new_full_name",
+    "up": "(SELECT CASE WHEN full_name IS NULL OR LENGTH(full_name) <= 3 THEN 'placeholder' ELSE full_name END)",
+    "down": "full_name"
+  }
+}]
+```
+```jsonc
+[{
+  "drop_table": {
+    "name": "clients"
+  }
+}]
+```
+```jsonc
+[{
+  "drop_column": {
+    "table": "clients",
+    "column": "full_name",
+    "down": "''"
+  }
+}]
+```
+</TabbedCode>
 
 See the [pgroll Operations Reference](https://github.com/xataio/pgroll/blob/main/docs/README.md#operations-reference) for the list of supported migration operations.
+
+## Lifecycle
+
+A multi-version schema migration's life cycle consists of `start`, `complete`, and `rollback` steps.
+
+### Start Step
+
+After a migration is started it is considered `active`. During this time, further schema modifications are not possible. This is reflected in disabled buttons and inputs in the UI while a migration is ongoing. From there you can decide whether to `complete` or `rollback` the active migration.
```

Comment:
Can we have multiple `active` migrations in separate branches? If yes, maybe  we can replace 

"During this time, further schema modifications are not possible"

with 

"One one `active` migration per branch is possible."

Reply:
Makes sense, updated the copy. Thanks!